r2-netty/src/main/java/com/linkedin/r2/netty/client/HttpNettyClient.java

```diff
@@ -294,18 +302,23 @@ private void sendRequest(Request request, RequestContext requestContext, Map<Str
     // responsibility of invoking the callback is handed over to the pipeline.
     final Timeout<None> timeout = new Timeout<>(_scheduler, resolvedRequestTimeout, TimeUnit.MILLISECONDS, None.none());
     timeout.addTimeoutTask(() -> decoratedCallback.onResponse(TransportResponseImpl.error(
-        new TimeoutException("Exceeded request timeout of " + resolvedRequestTimeout + "ms"))));
+        new TimeoutException("Exceeded request timeout of " + resolvedRequestTimeout + "ms" +
+            (requestContext.getLocalAttr(R2Constants.REMOTE_SERVER_ADDR) == null ? " (timeout during DNS resolution)" : "")))));
 
     // resolve address
     final SocketAddress address;
     try
     {
       TimingContextUtil.markTiming(requestContext, TIMING_KEY);
+      _dnsResolutions.incrementAndGet();
+      long startTime = _clock.currentTimeMillis();
```

Comment:
Consider to use nanoTime() to be monotonic : https://stackoverflow.com/questions/2978598/will-system-currenttimemillis-always-return-a-value-previous-calls